core/include/detray/materials/interaction.hpp

```diff
@@ -50,6 +49,42 @@ struct interaction {
         return eps_per_length * running;
     }
 
+    DETRAY_HOST_DEVICE scalar_type derive_stopping_power(
+        const detray::material<scalar_type>& mat, const int /*pdg*/,
+        const scalar_type m, const scalar_type qOverP, const scalar_type q) {
+
+        const relativistic_quantities rq(m, qOverP, q);
+        const scalar_t Ne{mat.molar_electron_density()};
+        const scalar_t eps_per_length{rq.compute_epsilon_per_length(Ne)};
+        if (eps_per_length <= 0.f) {
+            return 0.f;
+        }
+
+        const scalar_t I{mat.mean_excitation_energy()};
+        const scalar_t dhalf{rq.compute_delta_half(mat)};
+        const scalar_t u{rq.compute_mass_term(constant<scalar_t>::m_e)};
+        const scalar_t wmax{rq.compute_WMax(m)};
+        // uses RPP2023 eq. 34.5 scaled from mass stopping power to linear
+        // stopping power and multiplied with the material thickness to get a
+        // total energy loss instead of an energy loss per length. the required
+        // modification only change the prefactor which becomes identical to the
+        // prefactor epsilon for the most probable value.
+        const scalar_t running{math_ns::log(u / I) + math_ns::log(wmax / I) -
+                               2.f * rq.m_beta2 - 2.f * dhalf};
```

Comment:
Would it be possible to call the logarithm function only once?
```suggestion
        const scalar_t running{math_ns::log(u * wmax / (I * I)) -
                               2.f * (rq.m_beta2 - dhalf)};
```

Reply:
Make sense. Though it should be `2.f * (rq.m_beta2 + dhalf)` instread

Reply:
You are right